### DISCUSSION TITLE
How to Setup Aspire project for Multi Environment deployments for kubernetes (on-site servers 1. Dev 2. Staging 3.Prod)  (Maybe with Aspir8)??

### DISCUSSION THREAD

**Discussion by plppp2001** [microsoft/aspire#6566] [General]
Does anyone know of any examples and/or links on how to do this?   maybe somehow use appsettings.Development.json -> appsettings.Staging.json -> appsettings.Production.json

Anyone have any articles on how to Aspire Multi Environment deployments ?

**Comment by LeftTwixWand** [comment]
I need this example as well😐
What I found for now - [Mixing deployment](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/application-insights#mixed-deployment). I think, it would be possible to use `Aspire.Hosting.Azure.KeyVault` for different envs. 
But I still need example of how to configure database connection strings in different envs...

**Comment by Davilink** [comment]
After looking at the code of aspire you can use
`aspire run -- --environment=Staging` (or using the dotnet cli: `cd MyProject.AppHost && dotnet run -- --environment=Staging`

and then you will be able to use

```cs
using Microsoft.Extensions.Hosting;

var builder = DistributedApplication.CreateBuilder(args);
if(builder.Environment.IsStaging())
{
// do something specifig for this environment
}
```

and now it will read the appsetting.Staging.json instead of appsettings.Development.json
